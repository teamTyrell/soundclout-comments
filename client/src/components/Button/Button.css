.Button {
  outline: none;
  border-radius: 3px;
  cursor: pointer;
}

.Button--primary {
  background-color: #FF5500;
  color: white;
  border: none;
}

.Button--secondary {
  background-color: white;
  color: #333;
  border: 1px solid #e5e5e5;
}

.Button--secondary:hover {
  border-color: #ccc;
}

.Button--medium {
  font-size: 1.4rem;
  padding: 3px 11px 3px 10px;
}

.Button--small {
  font-size: 1.1rem;
  padding: 3px 9px 3px 8px;
}
