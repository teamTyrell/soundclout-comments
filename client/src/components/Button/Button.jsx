import React from 'react';
import './Button.css';

const Button = props => (
  <button
    id={ props.id && props.id }
    onMouseOver={ props.onMouseOver && props.onMouseOver }
    className={

      `Button
      ${ props.type
        ? `Button--${ props.type }`
        : `Button--primary` }

       ${ props.size
        ? `Button--${ props.size }`
        : `Button--medium` }
        ${ props.className && props.className }`

    }
  >
    { props.children }
  </button>
);

export default Button;