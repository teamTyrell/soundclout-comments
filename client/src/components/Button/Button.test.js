import React from 'react';
import renderer from 'react-test-renderer';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Button from './Button';

Enzyme.configure({ adapter: new Adapter() });

describe('<Button />', () => {

  it('renders <Button />', () => {

    const component = renderer.create(<Button />);

    expect(component.toJSON()).toMatchSnapshot();

  });

});