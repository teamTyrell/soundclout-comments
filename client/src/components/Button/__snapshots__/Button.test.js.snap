// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> renders <Button /> 1`] = `
<button
  className="Button
      Button--primary

       Button--medium"
/>
`;
